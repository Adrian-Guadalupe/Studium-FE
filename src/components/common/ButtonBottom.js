import styled from "styled-components";

const ButtonButtom = styled.button`
 background: ${props => props.loginCTA ? 'white' : '#2E71FD'};
 color: ${props => props.signupCTA ? 'white' : '#2E71FD'};
 border: ${props => props.loginCTA ? '1px solid #2E71FD' : 'none'};
 border-radius: 4px;
 padding: 12px;
 margin: 0 auto 1em auto;
 display: block;
 width: 80%;
 cursor: pointer;
 font-family: 'Source Sans Pro';

 &:hover{
 background: ${props => props.loginCTA ? '#2E71FD' : 'white'};
 color: ${props => props.signupCTA ? '#2E71FD' : 'white'};
 border: ${props => props.loginCTA ? 'none' : '1px solid #2E71FD' };  
}
`;

// export { Button };
export default ButtonButtom;