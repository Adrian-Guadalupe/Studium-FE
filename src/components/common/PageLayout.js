import React from 'react';
import styled from 'styled-components';
// import {Header} from './Header';

const Content = styled.main`
    max-width: 800px;
    margin: 60px auto 0 auto;
    /* padding: 0 16px; */
    box-sizing: border-box;
    font-family: '';

    h1, h2, h3, h4, h5, h6{
        font-family: '';
    }
    .getStarted{
        display: flex;
        width: 90%;
        margin: 0 auto;        
    }

    .h2Tags{
        text-align: center;
        font-size: 2em;
        line-height: 24px;
        font-family: 'Monstserrat';
        color: #010101;
    }

    .pTags{
        text-align: center;
        font-size: 1.4em;
    }
    @media(min-width: 370px){

}
`;

export function PageLayout({children}){
    return (
        <>
            {/* <Header /> */}
            <Content>
                {children}
            </Content>
        </>
    )
}