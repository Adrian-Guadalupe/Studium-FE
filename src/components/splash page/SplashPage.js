import React, { useState, useEffect } from "react";
import styled from "styled-components";
import NavBar from "../navigation/NavBar";
import Button from "../common/Button";
import ButtonBottom from "../common/ButtonBottom";
import { PageLayout } from "../common/PageLayout";
import { Input } from "../common/Input";
import Spinner from "../common/Spinner";

let timeout;

const Form = styled.form`
  width: 100%;
  max-width: 800px;
  background: white;
  padding: 16px;
  box-sizing: border-box;
  color: black;
  border-radius: 4px;
`;

const SplashPage = () => {
  const [email, setEmail] = useState({email: ''})
  const [loading, setLoading] = useState(false);

  function handleInputChange(e){
    e.persist();
    setEmail(state => ({
        ...state,
        [e.target.name]: e.target.value
    }));
}

  function handleSubmit(e) {
    e.preventDefault();
    setLoading(true);
    timeout = setTimeout(() => {
      setLoading(false);
    }, 1000);
  }

  useEffect(() => {
    return () => {
      if (timeout) {
        clearTimeout(timeout);
      }
    };
  }, []);

  return (
    <>
      <PageLayout>
        <NavBar />
        <img
          src={require("../common/header.svg")}
          style={{ width: "100%" }}
          alt="headerimage"
        />
        <h2 className="h2Tags">
          A place for students to create flashcards, share, and learn{" "}
        </h2>
        <p className="pTags">Master any subject, one success at a time.</p>

        <Form onSubmit={handleSubmit} className="getStarted">
        {loading ? <Spinner /> : 
        <>
          <Input 
          value={email.email}
          placeholder="Enter your email"
          type="text"
          name="email"
          onChange={handleInputChange}
           />
          <Button type="submit" small>
            Get Started
          </Button>
          </>
        }
        </Form>

        <div style={{ display: "flex" }}>
          <p>OR</p>
        </div>
        <img
          src={require("../common/sprinting.svg")}
          style={{ width: "100%" }}
          alt=""
        />
        <h2 className="h2Tags">You aren’t here to waste time, we know</h2>
        <p className="pTags">
          Studium uses spaced repitition to help you spend your time studying
          only the things you don’t know. This is the most effiicent and useful
          way to study.
        </p>
        <img
          src={require("../common/reading.svg")}
          style={{ width: "100%" }}
          alt=""
        />
        <h2 className="h2Tags">
          Share the important info from your massive textbooks
        </h2>
        <p className="pTags">
          Share the info you’ve pulled from your massive textbooks. Let friends
          share with you the info they’ve pulled. Together you can conquer this
          class.
        </p>
        <img
          src={require("../common/sitting.svg")}
          style={{ width: "100%" }}
          alt=""
        />
        <h2 className="h2Tags">Your textbooks are massive</h2>
        <p className="pTags">
          We know your textbook is over 1000 pages. Let’s help you get the most
          important info into online flashcards and start studying.
        </p>
        <ButtonBottom loginCTA>Log In</ButtonBottom>
        <ButtonBottom signupCTA>Sign Up</ButtonBottom>
      </PageLayout>
    </>
  );
};

export default SplashPage;
