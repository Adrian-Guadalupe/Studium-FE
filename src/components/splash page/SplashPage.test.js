import React from "react";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import SplashPage from "./SplashPage";
import ButtonBottom from "./SplashPage";
import PageLayout from "./SplashPage";
// import Form from "./SplashPage";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("Tests for the SplashPage", async () => {
  render(<SplashPage />);

  const h2Tag = screen.getByText(/A place for students to create flashcards, share, and learn/i);
  expect(h2Tag).toBeInTheDocument();

  // const pTags = screen.getByText(/Master any subject Master any subject, one success at a time./i);
  // expect(pTags).toBeInTheDocument();
});

it("renders log in button correctly", () => {
  const { getByTestId } = render(<ButtonBottom />);
  expect(getByTestId("loginButton")).toHaveTextContent("Log In");
});

it("renders sign up button correctly", () => {
  const { getByTestId } = render(<ButtonBottom />);
  expect(getByTestId("signupButton")).toHaveTextContent("Sign Up");
});

// it("get started button submits", () => {
//   const onSubmit = jest.fn();
//   const { getByTestId } = render(<Form onSubmit={onSubmit} />);
//   fireEvent.submit(getByTestId("form"));
//   expect(onSubmit).toHaveBeenCalled();
// });

it("matches snapshot", () => {
  const landingpage = renderer.create(<PageLayout />).toJSON();
  expect(landingpage).toMatchSnapshot();
});
