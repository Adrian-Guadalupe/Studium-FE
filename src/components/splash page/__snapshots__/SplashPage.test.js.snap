// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  background: white;
  color: #2E71FD;
  border: 1px solid #2E71FD;
  border-radius: 4px;
  display: block;
  width: 67px;
  height: 22px;
  font-size: 12px;
  cursor: pointer;
  font-weight: 600;
  font-family: 'Source Sans Pro';
}

.c3:hover {
  background: #2e71fd;
  color: white;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  padding: 20px 20px 0 20px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  width: 100%;
  box-shadow: 0px 5px 5px #c4c4c4;
}

.c2 {
  width: 72px;
  height: 26px;
  font-family: "Poppins",sans-serif;
  font-style: bold;
  font-weight: 500;
  font-size: 17px;
  line-height: 25px;
  color: #2e71fd;
}

.c6 {
  background: #2e71fd;
  color: white;
  border: 1px solid #2e71fd;
  border-radius: 4px;
  padding: 12px;
  margin-left: 6px;
  display: block;
  width: 100px;
  height: 40px;
  cursor: pointer;
  font-family: "Source Sans Pro";
}

.c6:hover {
  color: #2e71fd;
  background: white;
}

.c7 {
  background: white;
  color: #2E71FD;
  border: 1px solid #2E71FD;
  border-radius: 4px;
  padding: 12px;
  margin: 0 auto 1em auto;
  display: block;
  width: 80%;
  cursor: pointer;
  font-family: 'Source Sans Pro';
}

.c7:hover {
  background: #2E71FD;
  color: white;
  border: none;
}

.c8 {
  background: #2E71FD;
  color: white;
  border: none;
  border-radius: 4px;
  padding: 12px;
  margin: 0 auto 1em auto;
  display: block;
  width: 80%;
  cursor: pointer;
  font-family: 'Source Sans Pro';
}

.c8:hover {
  background: white;
  color: #2E71FD;
  border: 1px solid #2E71FD;
}

.c0 {
  max-width: 800px;
  margin: 0 auto 0 auto;
  box-sizing: border-box;
  font-family: '';
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-family: '';
}

.c0 .getStarted {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 90%;
  margin: 0 auto;
}

.c0 .h2Tags {
  text-align: center;
  font-size: 2em;
  line-height: 24px;
  font-family: 'Monstserrat';
  color: #010101;
}

.c0 .pTags {
  text-align: center;
  font-size: 1.4em;
}

.c0 .centeredOr {
  margin: 0 auto;
}

.c5 {
  padding: 4px 8px;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 1em;
  font-family: '';
  margin-bottom: 8px;
  width: 50%;
  box-sizing: border-box;
  height: 40px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c4 {
  width: 100%;
  max-width: 800px;
  background: white;
  padding: 16px;
  box-sizing: border-box;
  color: black;
  border-radius: 4px;
}

@media (min-width:370px) {

}

<main
  className="c0"
>
  <header
    className="c1"
  >
    <a
      className="c2"
    >
      Studium
    </a>
    <button
      className="c3"
      onClick={[Function]}
      type="button"
    >
      Log In
    </button>
  </header>
  <img
    alt="headerimage"
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "header.svg",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <h2
    className="h2Tags"
  >
    A place for students to create flashcards, share, and learn
  </h2>
  <p
    className="pTags"
  >
    Master any subject, one success at a time.
  </p>
  <form
    className="c4 getStarted"
    data-testid="form"
    onSubmit={[Function]}
  >
    <input
      className="c5"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email"
      type="text"
      value=""
    />
    <button
      className="c6"
      type="submit"
    >
      Get Started
    </button>
  </form>
  <p
    className="centeredOr"
  >
    OR
  </p>
  <img
    alt="character sprintinging"
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "sprinting.svg",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <h2
    className="h2Tags"
  >
    You aren’t here to waste time, we know
  </h2>
  <p
    className="pTags"
  >
    Studium uses spaced repitition to help you spend your time studying only the things you don’t know. This is the most effiicent and useful way to study.
  </p>
  <img
    alt="character reading"
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "reading.svg",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <h2
    className="h2Tags"
  >
    Share the important info from your massive textbooks
  </h2>
  <p
    className="pTags"
  >
    Share the info you’ve pulled from your massive textbooks. Let friends share with you the info they’ve pulled. Together you can conquer this class.
  </p>
  <img
    alt="character sitting"
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "sitting.svg",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <h2
    className="h2Tags"
  >
    Your textbooks are massive
  </h2>
  <p
    className="pTags"
  >
    We know your textbook is over 1000 pages. Let’s help you get the most important info into online flashcards and start studying.
  </p>
  <button
    className="c7"
    data-testid="loginButton"
    onClick={[Function]}
  >
    Log In
  </button>
  <button
    className="c8"
    data-testid="signupButton"
    onClick={[Function]}
  >
    Sign Up
  </button>
</main>
`;
